components/Ink/src/MDCInkView.m

```diff
@@ -103,30 +118,65 @@ - (void)setCustomInkCenter:(CGPoint)customInkCenter {
   self.inkLayer.customInkCenter = customInkCenter;
 }
 
-- (MDCInkLayer *)inkLayer {
-  return (MDCInkLayer *)self.layer;
+- (MDCLegacyInkLayer *)inkLayer {
+  return (MDCLegacyInkLayer *)self.layer;
 }
 
 - (void)startTouchBeganAnimationAtPoint:(CGPoint)point
                              completion:(MDCInkCompletionBlock)completionBlock {
-  [self.inkLayer spreadFromPoint:point completion:completionBlock];
+  if (self.usesLegacyInkRipple) {
+    [self.inkLayer spreadFromPoint:point completion:completionBlock];
+  } else {
+    self.startInkRippleCompletionBlock = completionBlock;
+    MDCInkLayer *inkLayer = [MDCInkLayer layer];
+    inkLayer.inkColor = self.inkColor;
+    inkLayer.maxRippleRadius = self.maxRippleRadius;
+    inkLayer.animationDelegate = self;
+
+    switch (self.inkStyle) {
+      case MDCInkStyleBounded:
+        self.clipsToBounds = YES;
+        break;
+      case MDCInkStyleUnbounded:
+        self.clipsToBounds = NO;
+        break;
+    }
+
+    inkLayer.opacity = 0;
+    inkLayer.frame = self.bounds;
+    [self.layer addSublayer:inkLayer];
+    [self.inkLayers addObject:inkLayer];
+    [inkLayer startAnimationAtPoint:point];
+    self.activeInkLayer = inkLayer;
```

Comment:
Regarding the comment above, it looks like your intent here is to track all of the active ink layers so that they can be updated on bounds changes and canceled when desired. Perhaps this could be simplified by removing `self.activeInkLayer` and solely making use of `self.inkLayers`? This would ensure that all active ink animations stay up-to-date rather than just the most recent one (I'm assuming this is the desired behavior).

Reply:
activeInkLayer is used by the MDCInkLayerDelegate methods. The intent is that the delegate methods should only be triggered by the most recently triggered instance of the ink ripple, so the start and end completion blocks are only called once (rather than for each ink ripple).

Reply:
Rather than keeping track of all ink layers in a separate array, I've revised to access sublayers directly when layoutSubviews or cancelAllAnimationsAnimated is called.